api/handlers/datasources.py

```diff
@@ -0,0 +1,50 @@
+import database.database
+from fastapi import APIRouter, Depends, Response
+from pydantic import BaseModel
+
+router = APIRouter()
+
+
+class Datasource(BaseModel):
+    name: str
+    accountName: str
+    containerName: str
+    description: str
+
+
+class GetDatasourcesResponse(BaseModel):
+    datasources: list[Datasource]
+
+
+@router.post("/api/datasources", status_code=201)
+def create_datasource(
+    datasource: Datasource,
+    response: Response,
+    db: object = Depends(database.database.db),
+):
+    """
+    Create a new datasource. The datasource will be henceforth identified by accountName_containerName which
+    must be unique or this function will return a 400.
+    """
+    if db.datasources.find_one(
+        {
+            "accountName": datasource.accountName,
+            "containerName": datasource.containerName,
```

Comment:
Would not make senbse to add a field datasource_id that way makes a lot easier to search and link the metadta with the detasource later without the need to be converting and spliting things.

Reply:
Break down is required for the browse datasources page. It would make more sense to not have encoded meaning in the datasource_id have have the accountName/containerName in the URL specifically IMHO

Reply:
replaced datasource_id with explicit accountName and containerName